os_traits/const.py

```diff
@@ -67,9 +67,55 @@
 # ref: https://en.wikipedia.org/wiki/AMD-V
 HW_CPU_X86_SVM = _CPU_X86_NS + 'svm'
 
+# All hypervisor specific features are prefixed with this namespace
+_VIRT_NS = 'virt:'
+
+# All libvirt specific features are prefixed with this namespace
+_VIRT_LIBVIRT_NS = _VIRT_NS + 'libvirt:'
+
+# All Hyper-V specific features are prefixed with this namespace
+_VIRT_HYPERV_NS = _VIRT_NS + 'hyperv:'
+
+_VIRT_HYPERV_VM_GENERATION = _VIRT_HYPERV_NS + 'vm_generation:'
+VIRT_HYPERV_VM_GENERATION_1 = -_VIRT_HYPERV_VM_GENERATION + '1'
+VIRT_HYPERV_VM_GENERATION_2 = -_VIRT_HYPERV_VM_GENERATION + '2'
+
+VIRT_HYPERV_SECURE_BOOT = _VIRT_HYPERV_NS + 'secure_boot'
+VIRT_HYPERV_SHIELDED_VMS =_VIRT_HYPERV_NS + 'shielded_vms'
+
+# RemoteFX suffix
+_VIRT_HYPERV_REMOTEFX = _VIRT_HYPERV_NS + 'remotefx:'
+
+# VRAM is requested in MB
+_VIRT_HYPERV_REMOTEFX_MEMORY = _VIRT_HYPERV_REMOTEFX + 'vram:'
+VIRT_HYPERV_REMOTEFX_MEMORY_64 = _VIRT_HYPERV_REMOTEFX_MEMORY + '64'
+VIRT_HYPERV_REMOTEFX_MEMORY_128 = _VIRT_HYPERV_REMOTEFX_MEMORY + '128'
+VIRT_HYPERV_REMOTEFX_MEMORY_256 = _VIRT_HYPERV_REMOTEFX_MEMORY + '256'
+VIRT_HYPERV_REMOTEFX_MEMORY_512 = _VIRT_HYPERV_REMOTEFX_MEMORY + '512'
+VIRT_HYPERV_REMOTEFX_MEMORY_1024 = _VIRT_HYPERV_REMOTEFX_MEMORY + '1024'
```

Comment:
Totally agree that vram is a quantitative resource, but at the moment, you cannot request different quantities than those present here (and the list might be extended in the future).

So, as far as traits / features / capabilities go, as a user, I would like to know these things before making a wrong request. Plus, I'd want in the future that nova will be able to quickly reject a request that needs another vram value.

This was the reasoning behind adding this as a trait. IMO, vram amount, in the case of hyper-v, can be considered as both trait, and consumable resource.